### PR TITLE
update all dependencies and use xcaddy for build

### LINKED ISSUE #83: Error during parsing: unrecognized directive: s3browser
Hi,

I've built caddy 2 from source via xcaddy with the following command: 
`./go/bin/xcaddy build --output ./caddy --with github.com/techknowlogick/caddy-s3browser --with github.com/caddy-dns/cloudflare`

Here's the redacted Caddyfile I've been testing with:
```
example-domain.com {
                s3browser {
                        endpoint nyc3.digitaloceanspaces.com
                        region nyc3
                        key <key>
                        secret <secret>
                        bucket <bucket>
                        debug true
                }
}
```

When I try to validate the file to ensure it will work, I get the following error: 

> Error during parsing: unrecognized directive: s3browser

Any ideas? What may I be missing to get this going?

### PR DESCRIPTION
**Pull request by rossmeier**
xcaddy is the new build tool for caddy v2. Use that in all our builds to make
the build configs easier to understand and maintain in the future.
Fixes: #83

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -62,18 +62,11 @@ steps:
...
-    - mkdir -p /go/src/github.com/caddyserver/caddy
-    - cd /go/src/github.com/caddyserver/caddy
-    - wget https://raw.githubusercontent.com/caddyserver/caddy/v2.0.0/cmd/caddy/main.go
-    - sed -i '/plug in Caddy modules here/a _ "github.com/techknowlogick/caddy-s3browser"' main.go
-    - go mod init caddy
-    - go get github.com/caddyserver/caddy/v2@v2.0.0
-    - go mod edit -replace github.com/techknowlogick/caddy-s3browser=/drone/src
-    - go install -v .
-    - /go/bin/caddy version
+    - xcaddy build --with github.com/techknowlogick/caddy-s3browser=/drone/src
```
Comment (techknowlogick) [inline .drone.yml]:
```suggestion
    - xcaddy build --output=/usr/bin/caddy --with github.com/techknowlogick/caddy-s3browser=/drone/src
```

**Comment by rossmeier** [comment]
Oh, sorry for that force push. Didn't realize that you pushed

**Comment by techknowlogick** [comment]
@veecue oh no worries. I was just poking at drone as I suspect it doesn't have the same env vars as the caddy builder itself. It was a minor change that I can quickly make again :)

Thanks for creating this PR

**Review by techknowlogick — APPROVED** [review]

**Comment by rossmeier** [comment]
can you check if the docker build from the docker file works manually? Not 100% sure about that

**Comment by rossmeier** [comment]
Ok, just checked, Dockerfile should work fine

**Pull request merged**